inst/examples/RcppProgressETA/src/eta_progress_bar.hpp

```diff
@@ -19,141 +19,168 @@
-		  time(&end);
-		} else if ((int) (((double)_ticks_displayed / (double)_max_ticks) * 100) == 10 * _timer_count) {
-
-      _timer_count++;
-
-		  time(&end);
-
-		  double pas_time = std::difftime(end, start);
-		  double rem_time = (pas_time / progress) * (1 - progress);
-		  int pas_time_simple = (int) pas_time;
-      int rem_time_simple = (int) rem_time;
-
-      std::stringstream rem_strs;
-      if (rem_time_simple > 3600) {
-        rem_strs << (int) (rem_time_simple / 3600) << "h";
-      } else if (rem_time_simple > 60) {
-        rem_strs << (int) (rem_time_simple / 60) << "min";
-      } else {
-        rem_strs << rem_time_simple << "s";
+  public: // ====== LIFECYCLE =====
+    
+    /**
+    * Main constructor
+    */
+    ETAProgressBar()  {
+      _max_ticks = 50;
+      _ticks_displayed = 0;
+      _finalized = false;
+      _mode_flag = 1;
+      _timer_flag = true;
+    }
+    
+    ~ETAProgressBar() {
+    }
+    
+    public: // ===== main methods =====
+      
+      void display() {
+        REprintf("0%%   10   20   30   40   50   60   70   80   90   100%%\n");
+        REprintf("[----|----|----|----|----|----|----|----|----|----|\n");
+        flush_console();
       }
-      std::string rem = rem_strs.str();
-
-      std::stringstream pas_strs;
-      if (pas_time_simple > 3600) {
-        pas_strs << (int) (pas_time_simple / 3600) << "h";
-      } else if (pas_time_simple > 60) {
-        pas_strs << (int) (pas_time_simple / 60) << "min";
-      } else {
-        pas_strs << pas_time_simple << "s";
+      
+      // will finalize display if needed
+      void update(float progress) {
+        
+        // condition to manage time measurement
+        if (_timer_flag) {
+          _timer_flag = false;
+          time(&start);
+        } else if ((progress > 0.001) & !_finalized) {
+          
+          time(&end);
+          
+          // calculate passed time and remaining time (in seconds)
+          double pas_time = std::difftime(end, start);
+          double rem_time = (pas_time / progress) * (1 - progress);
+          
+          // convert seconds to time string
+          std::string time_string = _time_to_string(rem_time);
+          
+          // create progress bar string 
+          std::string progress_bar_string = _current_ticks_display(progress);
+          
+          // ensure overwriting of old time info
+          int empty_length = time_string.length();
+          std::string empty_space = std::string(empty_length, ' ');
+          
+          // merge progress bar and time string
+          std::stringstream strs;
+          strs << "|" << progress_bar_string << "| " << time_string << empty_space;
+          std::string temp_str = strs.str();
+          char const* char_type = temp_str.c_str();
+          
+          // print: remove old and replace with new
+          REprintf("\r");
+          REprintf("%s", char_type);
+          REprintf("\r");
```

Comment:
not sure about this one. Which is its purpose ?

Reply:
The second call of `REprintf("\r");`? I also wondered... @coatless, would you mind explaining?

Reply:
I was just being overly caution here I think...